apps/ember-test-app/tests/unit/validators/phone-test.ts

```diff
@@ -0,0 +1,607 @@
+import { setOwner } from '@ember/application';
+import { tracked } from '@glimmer/tracking';
+import { bind } from '@nrg-ui/ember/helpers/bind';
+import {
+  validator as buildValidator,
+  PhoneValidator,
+} from '@nrg-ui/ember/validation';
+import { setupIntl } from 'ember-intl/test-support';
+import { setupTest } from 'ember-test-app/tests/helpers';
+import { module, test } from 'qunit';
+
+import type { TestContext as TC } from '@ember/test-helpers';
+import type { Binding } from '@nrg-ui/ember';
+
+class Model {
+  @tracked
+  field?: string | null;
+}
+
+declare type TestContext = {
+  binding: Binding<Model>;
+  model: Model;
+} & TC;
+
+module('Unit | Validator | phone', function (hooks) {
+  setupTest(hooks);
+  setupIntl(hooks, 'en-us');
+
+  hooks.beforeEach(function (this: TestContext) {
+    this.model = new Model();
+    this.binding = bind(this.model, 'field');
+
+    setOwner(this.model, this.owner);
+  });
+
+  test('`allowBlank` option works', function (this: TestContext, assert) {
+    const validator = new PhoneValidator(
+      this.binding,
+      { allowBlank: true },
+      this.model,
+    );
+
+    this.model.field = '';
+    let result = validator.result;
+
+    assert.deepEqual(result, {
+      isValid: true,
+      isWarning: false,
+      message: undefined,
+    });
+
+    this.model.field = null;
+    result = validator.result;
+
+    assert.deepEqual(result, {
+      isValid: true,
+      isWarning: false,
+      message: undefined,
+    });
+
+    this.model.field = undefined;
+    result = validator.result;
+
+    assert.deepEqual(result, {
+      isValid: true,
+      isWarning: false,
+      message: undefined,
+    });
+  });
+
+  test('`areaCode` option works', function (this: TestContext, assert) {
+    let validator = new PhoneValidator(this.binding, {}, this.model);
+
+    this.model.field = '5557890';
+
+    let result = validator.result;
+
+    assert.deepEqual(result, {
+      isValid: true,
+      isWarning: false,
+      message: undefined,
+    });
```

Comment:
Can we wrap this in a function to aid readability and general dryness?
`assertValid(validator)`, `assertInvalid(validator, message)`, `assertWarning(validator, message)`

I find myself breaking flow to read each of the options passed to `deepEqual` each time.

Reply:
Can I do that in a separate PR? That's a change I'd like to make in all the validator tests